payjoin-directory/src/lib.rs

```diff
@@ -440,6 +451,24 @@ async fn get_ohttp_keys(
     Ok(res)
 }
 
+async fn get_ohttp_allowed_purposes() -> Result<Response<BoxBody<Bytes, hyper::Error>>, HandlerError>
```

Comment:
This doesn't produce an error. Have you considered returning the plain `Response` and calling 

`"allowed-purposes" => Ok(get_ohttp_allowed_purposes().await),`

in `handle_ohttp_gateway_get` instead? Or did you want this to return a `Result` for consistency?